setup.py

```diff
@@ -14,29 +14,31 @@
 
 PACKAGE_NAME = os.getenv('QISKIT_AER_PACKAGE_NAME', 'qiskit-aer')
 _DISABLE_CONAN = distutils.util.strtobool(os.getenv("DISABLE_CONAN", "OFF").lower())
+_DISABLE_DEPENDENCY_INSTALL = distutils.util.strtobool(os.getenv("DISABLE_DEPENDENCY_INSTALL", "OFF").lower())
+
+if not _DISABLE_DEPENDENCY_INSTALL:
```

Comment:
It is probably better to move this under the `except` in each case with a statement like
```
dependency = 'conan>=1.31.2'
if _DISABLE_DEPENDENCY_INSTALL:
    print("{} is a required dependency. Please provide it and repeat install".format(dependency))
    sys.exit(1)
else:
    subprocess.call([sys.executable, '-m', 'pip', 'install', dependency])
    from conans import client
```